src/providers/ipa/ipa_deskprofile_rules_util.c

```diff
@@ -0,0 +1,917 @@
+        ret = ENOMEM;
+        goto done;
+    }
+
+    *_user_prio = talloc_steal(mem_ctx, user_prio);
+    *_group_prio = talloc_steal(mem_ctx, group_prio);
+
+    ret = EOK;
+
+done:
+    talloc_free(tmp_ctx);
+    return ret;
+}
+
+static errno_t
+ipa_deskprofile_rule_check_memberhost(TALLOC_CTX *mem_ctx,
+                                      struct sss_domain_info *domain,
+                                      struct sysdb_attrs *rule,
+                                      const char *rule_name,
+                                      const char *rule_prio,
+                                      const char *base_dn,
+                                      const char *hostname,
+                                      char **_host_prio,
+                                      char **_hostgroup_prio)
+{
+    TALLOC_CTX *tmp_ctx;
+    struct ldb_dn *host_dn;
+    struct ldb_message_element *el_orig_memberof = NULL;
+    struct ldb_message_element *el = NULL;
+    struct ldb_message **msgs;
+    size_t count;
+    size_t num_memberhostgroup;
+    char **memberhostgroups = NULL;
+    char *data;
+    char *memberhost;
+    char *memberhostgroup;
+    char *name;
+    char *host_prio;
+    char *hostgroup_prio;
+    const char *memberof_attrs[] = { SYSDB_ORIG_MEMBEROF, NULL };
+    bool host = false;
+    bool hostgroup = false;
+    errno_t ret;
+
+    tmp_ctx = talloc_new(mem_ctx);
+    if (tmp_ctx == NULL) {
+        return ENOMEM;
+    }
+
+    host_dn = sysdb_custom_dn(tmp_ctx, domain, hostname,
+                              DESKPROFILE_HOSTS_SUBDIR);
+    if (host_dn == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    ret = sysdb_search_entry(tmp_ctx, domain->sysdb, host_dn,
+                             LDB_SCOPE_BASE, NULL,
+                             memberof_attrs,
+                             &count, &msgs);
+    if (ret == ENOENT || count == 0) {
+        memberhostgroups = talloc_array(tmp_ctx, char *, 1);
+        memberhostgroups[0] = NULL;
+    } else if (ret != EOK) {
+        goto done;
+    } else if (count > 1) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "More than one result for a BASE search!\n");
+        ret = EIO;
+        goto done;
+    } else { /* ret == EOK && count == 1 */
+        el_orig_memberof = ldb_msg_find_element(msgs[0], SYSDB_ORIG_MEMBEROF);
+        memberhostgroups = talloc_array(tmp_ctx,
+                                        char *,
+                                        el_orig_memberof->num_values);
+    }
+
+    if (el_orig_memberof != NULL) {
+        num_memberhostgroup = 0;
+        for (size_t i = 0; i < el_orig_memberof->num_values; i++) {
+            data = (char *)el_orig_memberof->values[i].data;
+
+            ret = ipa_common_get_hostgroupname(tmp_ctx, domain->sysdb, data,
+                                               &name);
+            if (ret != EOK && ret != ERR_UNEXPECTED_ENTRY_TYPE) {
```

Comment:
Here we skip all errors except `ERR_UNEXPECTED_ENTRY_TYPE` explicitly but later we just silently continue if ret is `ERR_UNEXPECTED_ENTRY_TYPE`, is that expected?